src/NServiceBus.FileBasedRouting/XmlRoutingFileAccess.cs

```diff
@@ -1,27 +1,23 @@
 ﻿namespace NServiceBus.FileBasedRouting
 {
     using System;
-    using System.IO;
     using System.Xml;
     using System.Xml.Linq;
 
     class XmlRoutingFileAccess
     {
-        public string FilePath { get; }
+        public Uri FileUri { get; }
 
-        public XmlRoutingFileAccess(string filePath)
+        public XmlRoutingFileAccess(Uri fileUri)
         {
-            FilePath = filePath;
+            FileUri = fileUri;
         }
 
         public XDocument Read()
         {
             try
             {
-                using (var fileStream = File.OpenRead(FilePath))
-                {
-                    return XDocument.Load(fileStream);
-                }
+                return XDocument.Load(FileUri.ToString());
```

Comment:
@timbussmann The only security issue that comes to my mind is [XXE Injection](https://depthsecurity.com/blog/exploitation-xml-external-entity-xxe-injection) that can be prevented by [replacing XXE Entities](http://stackoverflow.com/questions/33255286/xdocument-parse-avoid-replacing-xxe-references). Is this the scenario we're facing though? I don't think so.